presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewInformationExtractor.java

```diff
@@ -47,15 +47,6 @@
     @Override
     protected Void visitQuerySpecification(QuerySpecification node, Void context)
     {
-        if (node.getLimit().isPresent()) {
-            throw new SemanticException(NOT_SUPPORTED, node, "Limit clause is not supported in query optimizer");
-        }
-        if (node.getHaving().isPresent()) {
-            throw new SemanticException(NOT_SUPPORTED, node, "Having clause is not supported in query optimizer");
-        }
-        if (!node.getFrom().isPresent()) {
-            throw new SemanticException(NOT_SUPPORTED, node, "Materialized view with no From clause is not supported in query optimizer");
-        }
         materializedViewInfo.setBaseTable(node.getFrom().get());
```

Comment:
IIUC, these `isPresent()`are now in `MaterializedViewRewriteQueryShapeValidator.Visitor#validateMaterializedViewOptimizationQueryShape`. But I don't see `node.getFrom()` is checked and `get()` is called.

Reply:
We actually do the check in `MaterializedViewQueryOptimizer` now https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java#L147